src/kokkos_ext/ArborX_DetailsKokkosExtMemory.hpp

```diff
@@ -0,0 +1,35 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_DETAILS_KOKKOS_EXT_STD_MEMORY_HPP
+#define ARBORX_DETAILS_KOKKOS_EXT_STD_MEMORY_HPP
+
+#include <Kokkos_Macros.hpp>
+
+#include <utility>
+
+namespace ArborX::Details::KokkosExt
+{
+
+template <class T, class... Args>
+KOKKOS_INLINE_FUNCTION constexpr T *construct_at(T *p, Args &&...args)
```

Comment:
```suggestion
KOKKOS_FUNCTION constexpr T *construct_at(T *p, Args &&...args)
```

Reply:
Also noting that the constexpr specifier only makes sense with C++20 onwards

Reply:
Why do you not want inline here?

Reply:
Because it is not necessary, only a hint and the compiler will just do the right thing.
I see having it as noise.